packages/core/_src/stm/TMap/operations/get.ts

```diff
@@ -0,0 +1,29 @@
+import { concreteTArray } from "@effect/core/stm/TArray/operations/_internal/InternalTArray"
+import { TMap } from "@effect/core/stm/TMap/definition"
+import { concreteTMap } from "@effect/core/stm/TMap/operations/_internal/InternalTMap"
+
+/**
+ * Retrieves value associated with given key.
+ *
+ * @tsplus fluent ets/TMap get
+ */
+export function get_<K, V>(self: TMap<K, V>, k: K): USTM<Option<V>> {
+  concreteTMap(self)
+  return STM.Effect((journal) => {
+    const buckets = self.tBuckets.unsafeGet(journal)
+
+    concreteTArray(buckets)
+
+    const idx = TMap.indexOf(k, buckets.chunk.length)
+    const bucket = buckets.chunk.unsafeGet(idx)!.unsafeGet(journal)
+
+    return bucket.find((_) => _.get(0) === k).map((_) => _.get(1))
```

Comment:
I think we may want to use `Equals.equals` for the equality check here since Scala is doing `==` which doesn't technically use reference equality.